<a href="https://colab.research.google.com/github/Perlangas28/Simulacion_1/blob/main/Bootstrap.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Bootstrap**

**Alumna: Perla Blancas Hernández**  
**Materia:** Simulación 1
**Prueba aplicada:** Bootstrap

Esta es una muestra aleatoria de $12$ tiempos de entrega:

<p align="center">$35, 42, 38, 40, 45, 37, 39, 41, 44, 36, 43, 40$</h1>

Instrucciones:

1. Use la muestra original de $12$ observaciones y genere $1000$ muestras Bootstrap.

2. Para cada muestra $B$ calcule la media

3. Al final:

- Obtenga la media de las $1000$ muestras.
- Calcule la desviación estándar.
- Determine los percentiles $2.5$% y $97.5$% para construir el interbalo de confianza $B$ del $95$%

In [1]:
import numpy as np

Primero guardamos los 12 tiempos de entrega en un arreglo y definimos

$n =  12$  observaciones

$B = 1000$  remuestreos Bootstrap

In [13]:
datos = np.array([35, 42, 38, 40, 45, 37, 39, 41, 44, 36, 43, 40])

n = len(datos)
B = 1000

Ahora generamos $1000$ muestras Bootstrap. Utilizaremos $replace=True$ para que un dato pueda repetirse varias veces dentro de una misma muestra. Además calculamos media, desviación estándar y percentiles para IC del $95$%

In [14]:
medias_bootstrap = []
for i in range (B):
  muestra_boot = np.random.choice(datos, size=n, replace=True)
  media_boot = np.mean(muestra_boot)
  medias_bootstrap.append(media_boot)

medias_bootstrap = np.array(medias_bootstrap)

media_bootstrap = np.mean(medias_bootstrap)
desv_bootstrap = np.std(medias_bootstrap, ddof=1)

percentil_2_5 = np.percentile(medias_bootstrap, 2.5)
percentil_97_5 = np.percentile(medias_bootstrap, 97.5)

print("Media muestra original= ", np.mean(datos))
print("Media de las 1000 muestras Bootstrap= ", media_bootstrap)
print("Desviación estándar Bootstrap= ", desv_bootstrap)
print("Percentil 2.5%= ", percentil_2_5)
print("Percentil 97.5%= ", percentil_97_5)
print("Intervalo de confianza Bootstrap 95%= (", percentil_2_5, ",", percentil_97_5, ")")




Media muestra original=  40.0
Media de las 1000 muestras Bootstrap=  40.0375
Desviación estándar Bootstrap=  0.8982223171959639
Percentil 2.5%=  38.25
Percentil 97.5%=  41.666666666666664
Intervalo de confianza Bootstrap 95%= ( 38.25 , 41.666666666666664 )


Como conclusión, el tiempo promedio de entrega estimado es aproximadamente 40 unidades de tiempo, y con un 95% de confianza el promedio poblacional se encuentra entre 38.25 y 41.58. El error estándar Bootstrap obtenido es aproximadamente 0.89.